lib/aliquot-pay/util.rb

```diff
@@ -11,18 +11,18 @@ def self.generate_shared_secret(private_key, public_key)
       private_key.dh_compute_key(public_key)
     end
 
-    def self.derive_keys(ephemeral_public_key, shared_secret, info)
+    def self.derive_keys(ephemeral_public_key, shared_secret, info, length: 32)
       input_keying_material = ephemeral_public_key + shared_secret
       if OpenSSL.const_defined?(:KDF) && OpenSSL::KDF.respond_to?(:hkdf)
         h = OpenSSL::Digest::SHA256.new
-        hbytes = OpenSSL::KDF.hkdf(input_keying_material, hash: h, salt: '', length: 32, info: info)
+        hbytes = OpenSSL::KDF.hkdf(input_keying_material, hash: h, salt: '', length: length * 2, info: info)
       else
-        hbytes = HKDF.new(input_keying_material, algorithm: 'SHA256', info: info).next_bytes(32)
+        hbytes = HKDF.new(input_keying_material, algorithm: 'SHA256', info: info).next_bytes(length * 2)
       end
 
       {
-        aes_key: hbytes[0..15],
-        mac_key: hbytes[16..32],
+        aes_key: hbytes[0..length - 1],
+        mac_key: hbytes[length..2 * length],
```

Comment:
Not `length..(2 * length) - 1` ? I.e. 32..63.

Reply:
Alternatively, `length..-1`

Reply:
Yes, it is. luckily it still performs as it should. Now I'm considering it, I prefer.
```ruby
        aes_key: hbytes[0, length],
        mac_key: hbytes[length, length],
```